beefy-primitives/src/commitment.rs

```diff
@@ -95,6 +96,130 @@ impl<TBlockNumber, TPayload, TSignature> SignedCommitment<TBlockNumber, TPayload
 	}
 }
 
+/// Type to be used to denote placement of signatures
+type BitField = Vec<u8>;
+/// Compress 8 bit values into a single u8 Byte
+const CONTAINER_BIT_SIZE: usize = 8;
+
+/// Temporary representation used for encoding efficiency.
+#[derive(Clone, Debug, PartialEq, Eq, Encode, Decode)]
+struct TemporarySignatures<TCommitment, TSignature> {
+	commitment: TCommitment,
+	signatures_from: BitField,
```

Comment:
Would be cool to add docs what every bit signifies and a more general description on the entire struct to cover how things get's packed.